src/internal/SignalGenerator.cpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file SignalGenerator.cpp
+ * @author Luciana Moreno Borges (lumoreno.borges@gmail.com)
+ * @brief Signal Generator class
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <bits/stdc++.h>
+#include <chrono>
+#include <hyro/SignalGenerator.h>
+
+namespace hyro
+{
+  /**
+   * @brief Construct a new Signal Generator:: Signal Generator object
+   * 
+   */
+SignalGenerator::SignalGenerator()
+{
+  m_deg = 0;
+}
+
+ 
+ /* Get the sine or cos for the desired angle */
+float SignalGenerator::getSignalValue(bool cosine)
```

Comment:
IMO, boolean flags is something to be avoided. it's not clear what to expect from `getSignalValue( true )` without looking into the implementation.

Better would be to pass an enun as argument:

```
enum Funtion {cosine, sine};

double SignalGenerator::getSignalValue(Funtion func)
{
...
if (func == cosine)
  return m_amp*cos(rad);
...
}
```

IMO, it's better to read `getSignalValue( cossine )`.